### PR TITLE
Static Graph Counter

### PR DESCRIPTION
**Pull request by mbrookhart**

### CODE REVIEW

File: src/ngraph/ngraph_compiler.cc [hunk 1]
```diff
@@ -109,16 +112,25 @@ void Compiler::Infer(const SimpleBindArg* simplebind) {
   }
 }
 
+static std::atomic<int> graph_counter(0);
+
+std::string get_ngraph_name() {
+  std::stringstream name;
+  name << "ngraph_" << std::this_thread::get_id() << "_" << graph_counter;
```
Comment (louisfeng) [inline src/ngraph/ngraph_compiler.cc]: This is not thread safe even with the `atomic` data type. You could do something like:
`name << "ngraph_" << std::this_thread::get_id() << "_" << graph_counter++;` and remove line 120.

Comment (louisfeng) [reply src/ngraph/ngraph_compiler.cc]: I also wonder if it's necessary to have thread id in the name if the `graph_counter` is already atomic and thread safe. The thread id will be potentially different from run to run.

Comment (mbrookhart) [reply src/ngraph/ngraph_compiler.cc]: The main reason for the thread name is that sockeye dlopen's libmxnet multiple times in multiple threads, so with just a counter, it's not unique

**Comment by mbrookhart** [comment]
TESTNOW

**Pull request merged**